ctapipe/io/serializer.py

```diff
@@ -0,0 +1,426 @@
+"""
+Serialize ctapipe containers to file
+"""
+
+from astropy.table import Table, Column
+from ctapipe.core import Container
+from abc import ABC, abstractmethod
+from astropy import log
+from pickle import dump
+from pickle import load
+from traitlets import Unicode
+import numpy as np
+from gzip import open as gzip_open
+
+__all__ = ['FitsSerializer', 'PickleSerializer']
+
+
+class Serializer:
+    """
+    Serializes ctapipe.core.Component, write it to a file thanks
+    to its Writer object
+    For some formats (i.e. pickle +gzip), read serialized components from
+    a file
+    """
+
+    def __init__(self, filename, format='fits', mode='x'):
+
+        """
+        Parameters
+        ----------
+        filename: str
+            full path name for i/o file
+        format: str ('fits', 'img', 'pickle')
+        mode: str ('write', 'read')
+            : use this serializer as writer or reader
+        mode: str
+            'r'	open for reading
+            'w'	open for writing, truncating the file first
+            'x'	open for exclusive creation, failing if the file already exists
+            'a'	open for writing, appending to the end of the file if it exists
+        Raises
+        ------
+        NotImplementedError: when format is not implemented
+        ValueError: when mode is not correct
+        """
+        self.filename = filename
+        self.format = format
+
+        if mode not in ['r', 'x', 'w', 'a']:
+            raise ValueError('{} is not a valid write mode. Use x, w or a'.
+                             format(mode))
+        self._writer = None
+        self._reader = None
+
+        if self.format == 'fits':
+            self._writer = TableWriter(outfile=filename, mode=mode,
+                                       format=format)
+        elif self.format == 'pickle':
+            if mode == 'r':
+                self._reader = GZipPickleReader(infile=filename)
+            else:
+                self._writer = GZipPickleWriter(outfile=filename,
+                                                mode=mode)
+        elif self.format == 'img':
```

Comment:
I think this should be a general `else`, like this a completely unknown format string would not result in an error